airflow/models/connection.py

```diff
@@ -473,8 +473,18 @@ def get_connection_from_secrets(cls, conn_id: str) -> Connection:
                 if hasattr(sys.modules.get("airflow.sdk.execution_time.task_runner"), "SUPERVISOR_COMMS"):
                     # TODO: AIP 72: Add deprecation here once we move this module to task sdk.
                     from airflow.sdk import Connection as TaskSDKConnection
-
-                    return TaskSDKConnection.get(conn_id=conn_id)
+                    from airflow.sdk.exceptions import AirflowRuntimeError, ErrorType
+
+                    try:
+                        return TaskSDKConnection.get(conn_id=conn_id)
+                    except AirflowRuntimeError as e:
+                        if e.error.error == ErrorType.CONNECTION_NOT_FOUND:
+                            log.exception(
+                                "Unable to retrieve connection from MetastoreBackend using Task SDK"
+                            )
```

Comment:
Do we need the log here too? I don't think we had the equivalent of this before

Reply:
I like it, lets keep the log. Gives more info as to whats happening.

Reply:
My thought is that that should be the responsiblity of the caller generally? (I don't like  un-silenceable logs for errors I can choose to handle myself and turn not in to an error)

Reply:
How about changing it to `log.debug` instead?

Reply:
I like the compromise of a debug log here.

Reply:
@ashb Are you all good with this approach?

Reply:
Thanks i think debug log makes sense here too